main.go

```diff
@@ -61,22 +61,29 @@ func prettyPrintStruct(item interface{}) string {
 
 func startWatchdog(config types.Configuration) {
 	healthChan := make(chan types.StatsEvent)
-
+	client := getTelemetryClient(config)
+	
 	go routing.StartCheck(config, healthChan)
 	go health.StartCheck(config, healthChan)
 
 	for {
 		event := <-healthChan
-		publishToAppInsights(event, config)
+		publishToAppInsights(client, event, config)
 		if config.Debug {
 
 		}
 		fmt.Println(prettyPrintStruct(event))
 	}
 }
 
-func publishToAppInsights(event types.StatsEvent, config types.Configuration) {
-	client := appinsights.NewTelemetryClient(config.AppInsightsKey)
+func getTelemetryClient(config types.Configuration) appinsights.TelemetryClient {
```

Comment:
newTelemetryClient might be more explicit as the function is constructing a new client every call and does not cache one for retrieval.